lib/chef/knife/ec2_server_create.rb

```diff
@@ -1263,22 +1350,47 @@ def evaluate_node_name(node_name)
         node_name % server.id
       end
 
-      def create_volume_tags(hashed_volume_tags)
-        hashed_volume_tags.each_pair do |key, val|
-          ec2_connection.tags.create key: key, value: val, resource_id: @server.block_device_mapping.first["volumeId"]
-        end
-      end
       # TODO: connection_protocol and connection_port used to choose winrm/ssh or 5985/22 based on the image chosen
       def connection_port
         port = config_value(:connection_port,
                             knife_key_for_protocol(connection_protocol, :port))
-        port || winrm? ? 5985 : 22
+        return port if port
+
+        assign_default_port
+      end
+
+      # Set default port 5986 if winrm_ssl return true otherwise set it to 5985
+      # Set default port 22 for ssh protocol
+      # @return [Integer]
+      def assign_default_port
```

Comment:
How about using such elements by a constant.

`DEFAULT_PORT = winrm? ? config_value(:winrm_ssl) ? 5985 : 5986 : 22`

Reply:
In my openion, As we required these value at instance level so not recommended. Thanks